homeassistant/components/netatmo/config_flow.py

```diff
@@ -45,11 +65,113 @@ def extra_authorize_data(self) -> dict:
 
     async def async_step_user(self, user_input=None):
         """Handle a flow start."""
-        if self.hass.config_entries.async_entries(DOMAIN):
-            return self.async_abort(reason="already_setup")
-
+        await self.async_set_unique_id(DOMAIN)
         return await super().async_step_user(user_input)
 
     async def async_step_homekit(self, homekit_info):
         """Handle HomeKit discovery."""
         return await self.async_step_user()
+
+
+class NetatmoOptionsFlowHandler(config_entries.OptionsFlow):
+    """Handle Netatmo options."""
+
+    def __init__(self, config_entry: config_entries.ConfigEntry):
+        """Initialize Netatmo options flow."""
+        self.config_entry = config_entry
+        self.options = dict(config_entry.options)
+        self.options.setdefault(CONF_WEATHER_AREAS, {})
+
+    async def async_step_init(self, user_input=None):
+        """Manage the Netatmo options."""
+        return await self.async_step_public_weather_areas()
+
+    async def async_step_public_weather_areas(self, user_input=None):
+        """Manage configuration of Netatmo public weather areas."""
+        errors = {}
+
+        if user_input is not None:
+            new_client = user_input.pop(CONF_NEW_AREA, None)
+            areas = user_input.pop(CONF_WEATHER_AREAS, None)
+            user_input[CONF_WEATHER_AREAS] = {
+                area: self.options[CONF_WEATHER_AREAS][area] for area in areas
+            }
+            self.options.update(user_input)
```

Comment:
How are areas removed from the options? It seems we're only adding areas and never removing.

Reply:
The dictionary is updated according to the selected areas of the multi select field. See L109-116

So, if you don't need an area any more, you simply deselect it in the multi select field and it will be removed once the changes are submitted.

Reply:
Sorry, I missed that the areas were nested under a single key that we update.

Reply:
No problem. I am not particularly happy with the UX but I think right now there are no better alternatives.